File: src/main/java/com/windchopper/fs/sftp/SftpFileAttributesView.java
package com.windchopper.fs.sftp;

import java.nio.file.attribute.BasicFileAttributeView;
import java.nio.file.attribute.BasicFileAttributes;
import java.nio.file.attribute.FileTime;

public class SftpFileAttributesView implements BasicFileAttributeView {

    final SftpFile file;

    SftpFileAttributesView(SftpFile file) {
        this.file = file;
    }

    @Override
    public String name() {
        return getClass().getSimpleName();
    }

    @Override
    public BasicFileAttributes readAttributes() {
        return new SftpFileAttributes(file.attributes);
    }

    @Override
    public void setTimes(FileTime lastModifiedTime, FileTime lastAccessTime, FileTime createTime) {
        throw new UnsupportedOperationException();
    }

}

File: src/main/java/com/github/windchopper/fs/sftp/SftpFileAttributesView.kt
package com.github.windchopper.fs.sftp

import java.nio.file.attribute.BasicFileAttributeView
import java.nio.file.attribute.BasicFileAttributes
import java.nio.file.attribute.FileTime

class SftpFileAttributesView internal constructor(val file: SftpFile): BasicFileAttributeView {

    override fun name(): String {
        return javaClass.simpleName
    }

    override fun readAttributes(): BasicFileAttributes {
        return SftpFileAttributes(file.attributes)
    }

    override fun setTimes(lastModifiedTime: FileTime, lastAccessTime: FileTime, createTime: FileTime) {
        TODO("not implemented")
    }

}